go/mysql/collations/collation.go

```diff
@@ -176,14 +176,14 @@ func register(c Collation) {
 	globalAllCollations[c.ID()] = c
 }
 
-func ToUpper(id ID, raw []byte) []byte {
+func ToUpper(id ID, raw []byte, dst []byte) []byte {
```

Comment:
This API is not necessary. You're not supposed to use `globalAllCollations` because some MySQL versions don't support _some_ of the collations. That's why we only expose these methods from the `Collation` interface.